keps/0005-cluster-resources-for-crds.md

```diff
@@ -85,25 +88,49 @@ should probably be created
+
+#### Instance Status update
+
+If Instances of an OperatorVersion are created before the Install plan of the OperatorVersion is successful, the Instance's will remian in a `PENDING` status until the OperatorVersion is successful.
+
+#### Object Owners
 
+The objects created by the `installPlan` may exist on the cluster before the OperatorVersion is installed. This may be a result of another OperatorVersion requiring the same objects, or the objects being managed externally to KUDO. If so, the OperatorVersion will be appended to the `ownerReferences`. If the object does not exist, then KUDO will create the object with the OperatorVersion as the `ownerReference`, and additionally add an annotation to the object that it was created by KUDO: `kudo.dev/managed: true`. This will be used during cleanup to identify whether the object was created by KUDO or not.
 
+If the object already exists, but does not match the spec provided by the `installPlan`, the `installPlan` will fatally error and the OperatorVersion's status will be set to `FatalError`. We will not support updating/changing existing objects as part of an install since this may change how components not managed by this OperatorVersion would function.
+
+#### Uninstalls and Object Cleanup
+
+When an OperatorVersion is uninstalled, KUDO will remove the OperatorVersion from the `ownerReference`. If there are no more `ownerReferences`, it will look for the `kudo.dev/managed: true` annotation to identify if this object is no longer needed on the cluster. If that annotation is there, it will remove the object. If it is not there, then the object must have existed before installing any OperatorVersions and is managed by a component outside of KUDO, and will remain on the cluster.
+
+#### Specifications of Operator
+
+### Risks and Mitigations
 
 ## Graduation Criteria
 
-The MySQL Operator could be modified to deploy a central repo for backups and leverage those in each instance of MySQL
+The MySQL Operator could be modified to deploy a central repo for backups and leverage those in each instance of MySQL. Spark Operator will be able to install the required CRDs on OperatorVersion installataion that can be used by instances of the Spark Operator
 
 ## Implementation History
 
 * Initial KEP - 20190218
+* Proposal for Implementation - 20200124
 
-## Drawbacks [optional]
+## Drawbacks
 
 * More complicated Operator/OperatorVersion specs
 * Implications of OperatorVersion installation making MORE cluster level changes than just a CRD.
 * Security around leveraging the common component.  If deploying minio for backups, how do we ensure Instances don't restore someone else's data, or overwrite someone else's backup?
 * Shared Components: What happens when two different Operators require the same common object?
```

Comment:
That would be handled by having multiple ownerReferences on the common object, one for each OperatorVersion that utilizes it, right?